File: 2nd year/MPP/project-mpp/csharp-project/persistence/validators/GameValidator.cs
using System.Text.RegularExpressions;
using model;

namespace persistence.validators
{
    public class GameValidator : IValidator<Game>
    {
        public void Validate(Game game)
        {
            if (game == null)
            {
                throw new ValidationException("Entity not instance of game");
            }

            var nameMatch = Regex.Match(game.Name, "^[A-Z]");
            if (game.Name == null || !nameMatch.Success)
            {
                throw new ValidationException("Game name is invalid (null or malformed)");
            }
        }
    }
}

File: 2nd year/MPP/project-mpp/java-project/TriathlonPersistence/src/main/java/triathlon/persistence/validators/GameValidator.java
package triathlon.persistence.validators;

import triathlon.model.Game;

public class GameValidator implements Validator<Game> {
    @Override
    public void validate(Game game) throws ValidationException {
        if (game == null) {
            throw new ValidationException("triathlon.model.entities.Entity not instance of game");
        }
        if (game.getName() == null
                || game.getName().matches("^[A-Z]")) {
            throw new ValidationException("Game name is invalid (null or malformed)");
        }
    }
}
